public/app/core/services/backend_srv.ts

```diff
@@ -83,9 +86,26 @@ export class BackendSrv implements BackendService {
     this.internalFetch = this.internalFetch.bind(this);
     this.fetchQueue = new FetchQueue();
     this.responseQueue = new ResponseQueue(this.fetchQueue, this.internalFetch);
+
+    this.initGrafanaDeviceID();
+
     new FetchQueueWorker(this.fetchQueue, this.responseQueue, getConfig());
   }
 
+  private async initGrafanaDeviceID() {
+    if (config.buildInfo && config.buildInfo.edition === GrafanaEdition.OpenSource) {
```

Comment:
Magic of the typescript and this might be simplified to
```suggestion
    if (config.buildInfo?.edition === GrafanaEdition.OpenSource) {
```